examples/ordinance_gpt/parse_pdf.py

```diff
@@ -0,0 +1,74 @@
+"""Example on parsing an existing PDF file on-disk for ordinances."""
+from functools import partial
+
+import openai
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+
+from rex import init_logger
+from elm.base import ApiBase
+from elm.web.document import PDFDocument
+from elm.ords.llm import LLMCaller
+from elm.ords.services.openai import OpenAIService
+from elm.ords.utilities import RTS_SEPARATORS
+from elm.ords.process import validate_api_params
+from elm.ords.extraction.ordinance import OrdinanceExtractor
+from elm.ords.extraction.apply import extract_ordinance_values
+from elm.ords.services.provider import RunningAsyncServices as ARun
+from elm.ords.extraction.apply import (check_for_ordinance_info,
+                                       extract_ordinance_text_with_llm)
+
+
+if __name__ == '__main__':
+    init_logger('elm', log_level='INFO')
+
+    # download this from https://app.box.com/s/a8oi8jotb9vnu55rzdul7e291jnn7hmq
+    fp_pdf = 'Palo Alto Iowa.pdf'
+
+    for idx in range(10):
+        fp_txt_all = fp_pdf.replace('.pdf', f'_all_{idx}.txt')
+        fp_txt_clean = fp_pdf.replace('.pdf', f'_clean_{idx}.txt')
+        fp_ords = fp_pdf.replace('.pdf', f'_ords_{idx}.csv')
+
+        doc = PDFDocument.from_file(fp_pdf)
```

Comment:
Could consider defining this and `text_splitter` below outside the `for` loop. If needed, you can deepcopy the starting doc at the beginning of the loop (or even easier - just reset the metadata)

Reply:
i actually wanted this script to just be a single pdf parse job. the iter was leftover from my experiment and i forgot to clean it up. Removed the iter.